api/app/models/slot.scala

```diff
@@ -10,10 +23,169 @@ final case class Slot(
 
 object Slot {
   implicit val testFmt = Json.format[Slot]
+
+  def read(json: String): Option[Slot] = {
+    Json.fromJson[Slot](Json.parse(json)) match {
+      case JsSuccess(slot, _) => Some(slot)
+      case JsError(errors) =>
+        println("unable to deserialise slot from JSON", errors, json)
+        None
+    }
+  }
+}
+
+trait SlotStore {
+  def getAll(): Future[List[Slot]]
+  def get(id: String): Future[Option[Slot]]
+  def upsert(slotId: String, updatedSlot: Slot): Future[Unit] // depends on insert case handling
+
+  // DEV method to ensure index exists and has some dummy data
+  def populate(): Future[Unit]
+}
+
+object SlotStore {
+  def apply(lifecycle: ApplicationLifecycle)(implicit ec: ExecutionContext): SlotStore = {
+    val useRemoteStore = sys.env.get("USE_REMOTE_STORE").contains("true")
+
+    try {
+      val store = if (true) {
+        ElasticsearchStore(lifecycle)
+      } else {
+        MemoryStore()
+      }
+
+      store.populate()
+      store
+    } catch {
+      case e: Throwable =>
+        println("unable to start ES, falling back to MemoryStore", e.getMessage)
```

Comment:
Really minor, but should this be:

```suggestion
        println("unable to connect to ES, falling back to MemoryStore", e.getMessage)
```